Lib/ast.py

```diff
@@ -551,6 +553,671 @@ def __new__(cls, *args, **kwargs):
     type(...): 'Ellipsis',
 }
 
+# Large float and imaginary literals get turned into infinities in the AST.
+# We unparse those infinities to INFSTR.
+INFSTR = "1e" + repr(sys.float_info.max_10_exp + 1)
+
+def interleave(inter, f, seq):
+    """Call f on each item in seq, calling inter() in between.
+    """
+    seq = iter(seq)
+    try:
+        f(next(seq))
+    except StopIteration:
+        pass
+    else:
+        for x in seq:
+            inter()
+            f(x)
+
+class _Unparser:
+    """Methods in this class recursively traverse an AST and
+    output source code for the abstract syntax; original formatting
+    is disregarded. """
+
+    def __init__(self, tree, file = sys.stdout):
```

Comment:
I'm not sure about this IO object API, I dont see a need to IO object. Maybe we can construct an internal list and append that list whenever `fill` called and then return the merged version of list like `Untokenizer`

Reply:
As I said in the PR description, this PR only ports the code from one side to the other to make the PR easy to review and any other modification should be fine in other PRs. Also notice that this class is private so this is not user facing.

I propose to discuss what you propose in a different pull request.

Reply:
> I propose to discuss what you propose in a different pull request.

Sure